### ISSUE TITLE
CRAN task view proposal: Mixed models

### ISSUE THREAD

**Issue by bbolker** [cran-task-views/ctv#35]
## Description

Mixed models are a broad class of statistical models used to analyze data where observations can be assigned *a priori* to discrete groups, and where the parameters describing the differences between groups are treated as random variables. They are also referred to as multilevel, or hierarchical, models; longitudinal data are often analyzed in this framework.

We had already started on this task view before writing the proposal, so a [repo](https://github.com/bbolker/mixedmodels-taskview) and a [draft](https://github.com/bbolker/mixedmodels-taskview/blob/main/MixedModels.md) are in progress.

## Scope

The proposed task view would include packages for linear, generalized linear and nonlinear mixed model fitting (including convenience wrappers), model diagnostics and summaries, and inferential tasks (e.g. hypothesis testing, prediction and estimation). Packages for a variety of specialized data types and model structures (e.g.  censored or zero-inflated data, spatial models) are included, as well as packages containing data sets that are widely used in teaching and learning about mixed models.  This proposed TV would **only** cover models that incorporate continuous (usually although not always Gaussian) latent variables; this excludes packages that handle hidden Markov Models, finite (discrete) mixture models, latent Markov models, etc.. 

## Packages

 Please see the [current draft](https://github.com/bbolker/mixedmodels-taskview/blob/main/MixedModels.md). The current set of proposed [core packages](https://github.com/bbolker/mixedmodels-taskview/issues/6):

- lme4
- nlme
- brms
- MCMCglmm
- multilevelmod
- geepack
- lavaan
- broom.mixed
- DHARMa
- lmerTest
- glmmTMB
- blme
- rstanarm
- ordinal
- robustlmm

## Overlap

There is some overlap with the [Agriculture task](https://cran.r-project.org/web/views/Agriculture.html) view regarding mixed models that include a kinship or relatedness matrix. We could point to that resource or vice versa. There is overlap with the [Robust](https://cran.r-project.org/web/views/Robust.html) which also has a mixed models section, and with the [Bayesian Inference](https://cran.r-project.org/web/views/Bayesian.html) task view.  

## Maintainers

Ben Bolker would be the principal maintainer. Julia Piaskowski (and others) would co-maintain.

**Comment by zeileis** [comment]
_Feedback as CRAN Task View Editor:_ Wonderful, this is a great proposal. It's very clear what goes in and what not and it's certainly a highly-relevant relevant topic. I'm very much looking forward to having it on CRAN. Details:

* Currently, the "Links" section cannot have subsections. In the [Econometrics](https://CRAN.R-project.org/view=Econometrics) we simply have prefixes to the links.
* As sections are not numbered in the CRAN style I would probably not rely on three levels of headers but just on two.

_Feedback as author:_ In the additive models section i think it would be appropriate to add `gamlss`, `VGAM`, and `bamlss`. And the GLMM trees in the `glmertree` package could also be incorporated into the specialized models section. Disclaimer: I'm co-auhor of `bamlss` and `glmertree`.

**Comment by rociojoo** [comment]
@bbolker It looks like a great proposal, thank you! Relevant, well structured, and easy to navigate. 
The multinomial responses section is empty but I guess you'll fill it soon.

**Comment by zeileis** [comment]
Both `VGAM` and `R2BayesX` support multinomial responses and should support GAMs with random-effect terms.

**Comment by tuxette** [comment]
Thanks for the proposal that looks good for me too! A few very minor suggestions:
- I don't really get why `emmeans` is only cited in the Prediction subsection and not in Hypothesis testing. In addition, I would have placed that package in core (but it's a personal point of view so I guess that you can disagree);
- You have a remaining "FIXME" in the section "Multinomial responses";
- You have an unnecessary comma at the end of the section "influential data points". In addition, in this part, titles do not start with a capital letter whereas they do in other parts;
- I think that, as you suggested, you should cite relevant related TV.

**Comment by zeileis** [comment]
Great, this means that we have endorsement from three CRAN Task View Editors and you can officially move forward.

Ben @bbolker, I suggest that refine the document you already have based on our feedback and then let us know about it. Also, you can try to bring on a couple of further co-maintainers, ideally persons which can add a different view, be it in terms of methodology (e.g., someone from the more "specialized" models), ethnicity, geographic region, etc.

In terms of links to other task views: Once your new task view is ready, we can contact the maintainers of the other task views and suggest that they link back to your view and possibly reduce the discussion of mixed models in their view.

**Comment by jpiaskowski** [comment]
It doesn't look like [VGAM](https://cran.r-project.org/package=VGAM) solves or supports mixed models...am I missing something?

**Comment by bbolker** [comment]
Agreed: I can't find anything. Both Yee et al 2010 and Yee et al 2015 say that allowing mixed models would be a useful extension ...

---

Yee, Thomas. 2010. “VGLMs and VGAMs: An Overview for Applications in Fisheries Research.” Fisheries Research 101 (January): 116–26. https://doi.org/10.1016/j.fishres.2009.09.015.
Yee, Thomas W., Jakub Stoklosa, and Richard M. Huggins. 2015. “The VGAM Package for Capture-Recapture Data Using the Conditional Likelihood.” Journal of Statistical Software 65 (June): 1–33. https://doi.org/10.18637/jss.v065.i05.

**Comment by zeileis** [comment]
Sorry for the confusion. I thought Thomas had added it at some point but apparently not. Thanks for checking!

**Comment by jpiaskowski** [comment]
I'm happy to relocate kinship-augmented mixed models from Agriculture to this CTV once it is published and point folks the mixed model CTV.

**Comment by strengejacke** [comment]
Thanks for pointing to this proposal. I would suggest adding following packages:

* clmm
* mixor

In the section *Model presentation and prediction* I would add:

* parameters
* modelsummary

In the section *Prediction and estimation* I would add

* marginaleffects

The packages from Vincent are quite actively developed and cover *a lot* of models.

**Comment by bbolker** [comment]
The last three are added.

Is there a `clmm` package, or is it a function within `ordinal` (which is already listed)?  `mixor` has been archived (quite a while ago, on 2021-07-31).  (The [Environmetrics](https://cran.r-project.org/web/views/Environmetrics.html) task view does include archived packages (the "archived" tag appears to be autogenerated since the code just has `r pkg(...)` as usual), but I think they're not supposed to be included ...?

**Comment by strengejacke** [comment]
Yes, my fault, I think clmm() is the function from pkg ordinal. And I agree that archived pkgs should not be included in the task view.

**Comment by zeileis** [comment]
Since the re-launch of the CRAN Task View Initiative archived packages do not need to be removed from task views immediately. Instead one can allow a certain period of grace (currently 60 days) before an issue is created calling for action, if necessary removal from the task view (after 100 days). See <https://github.com/cran-task-views/Environmetrics/issues> for some examples.

As this policy is still new, we haven't fully automated everything and are still trying to refine the process.

In any case, archived packages should not be included in a new task view unless it is very likely that the packages will be unarchived very soon.

**Comment by bbolker** [comment]
As far as I know we've resolved all of the issues discussed above. Current draft is at https://github.com/bbolker/mixedmodels-taskview/blob/main/MixedModels.md . What's the next step?  Migrate that document to a new repo under this organization? Prepare an appropriate README?

**Comment by zeileis** [comment]
If possible, please clean up the repository a bit. I just had a quick glance and my impression was that not all files are needed going forward. (But what is still needed can remain.)

Then please make me a co-owner of the task view. Then I can streamline the README and repository information, the task view file, and transfer the repository to the `cran-task-views` group.

Finally, you can check whether everything is in order and ready for release.

When I have the "go" from you, I release everything on CRAN.

**Comment by zeileis** [comment]
Thanks @bbolker and apologies for the confusion regarding the transfer. For future reference: The co-admin settings are only possible for repositories hosted by organizations. Otherwise the easiest solution is to transfer the repository to one CRAN Task View Editor directly (typically me). This can be done via:

> Settings > Collaborators > Public Repository > Manage

and then

> Danger Zone > Transfer ownership

to "zeileis".

**Comment by zeileis** [comment]
I have completed the transfer and standardization of the task view now, see:

https://github.com/cran-task-views/MixedModels/

* I use our standard `README.md` template and tweaked a few formatting issues.
* In the intro paragraphs I added cross-references to the `Cluster` and `TimeSeries` task views which partially cover some of the relevant packages.
* I copied the Bioc link from the intro to the links section as well.
* I omitted the comment that `bamlss` relies on JAGS because this can only be used optionally. Usually, the package uses its own samplers.

Please check whether everything is ok for you.

_Last detail:_ The list of maintainers and GitHub contributors should be synchronized (Julia is missing in the latter). And I wouldn't list the maintainers as contributors in the first paragraph of the intro (all their names are prominently in the header).
